### ISSUE TITLE
False-positive RawObjectIteration for descendants of Traversable

### ISSUE THREAD

**Issue by weirdan** [vimeo/psalm#334]
```php
<?php

class C implements IteratorAggregate
{
    public function getIterator(): Iterator
    {
        return new ArrayIterator([]);
    }
}

function loopT(Traversable $coll): void
{
    foreach ($coll as $item) {}
}

function loopI(IteratorAggregate $coll): void
{
    foreach ($coll as $item) {}
}

loopT(new C);
loopI(new C);
```
Here foreach in `loopI` should not emit `RawObjectIteration`, as `IteratorAggregate` extends `Traversable`. Neither should this problem be emitted for anything that extends `IteratorAggregate` (e.g. `Doctrine\Common\Collections\Collection`). Psalm should consider entire list of interfaces implemented (including transitively implemented interfaces) and not complain about iteration over anything that is `Traversable`.
